src/H5Dpublic.h

```diff
@@ -1276,22 +1277,31 @@ H5_DLL herr_t H5Dwrite_chunk(hid_t dset_id, hid_t dxpl_id, uint32_t filters, con
  *          \p buf is the memory buffer containing the chunk read from
  *          the dataset in the file.
  *
- * \attention Exercise caution when using H5Dread_chunk() and
+ *          \p nalloc must be passed as a pointer to a variable holding the
+ *          allocated size, in bytes, of the memory buffer \p buf. On exit,
+ *          \p *nalloc is set to the buffer size needed to read the chunk, which
+ *          is the same as the size of the chunk on disk. If the value of
+ *          \p *nalloc passed in was insufficient to read the entire, chunk, no
+ *          data is read. \p buf may be passed as NULL as long as \p *nalloc
```

Comment:
Why does `nalloc` need to be set to zero when `buf ` is NULL?

Reply:
Now that I look at it, H5Tencode allows *nalloc > 0 with buf == NULL, so I'll change this.

Reply:
Ok, then I will allow this behaviour in the FORTRAN wrapper.